tx_list.go

```diff
@@ -286,9 +319,47 @@ func (tx *Tx) LKeys(bucket, pattern string, f func(key string) bool) error {
 		return ErrBucket
 	}
 	for key := range tx.db.ListIdx[bucket].Items {
+		if tx.CheckExpire(bucket, []byte(key)) {
+			continue
+		}
 		if end, err := MatchForRange(pattern, key, f); end || err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func (tx *Tx) ExpireList(bucket string, key []byte, ttl uint32) error {
+	if err := tx.checkTxIsClosed(); err != nil {
+		return err
+	}
+	if _, ok := tx.db.ListIdx[bucket]; !ok {
+		return ErrBucket
+	}
+	list, err := tx.db.ListIdx[bucket].LRange(string(key), 0, -1)
+	if err != nil {
+		return err
+	}
+	tx.db.ListIdx[bucket].TTL[string(key)] = ttl
+	tx.db.ListIdx[bucket].TimeStamp[string(key)] = uint64(time.Now().Unix())
+	ttls := strconv2.Int64ToStr(int64(ttl))
+	err = tx.push(bucket, key, DataExpireListFlag, []byte(ttls))
+	if err != nil {
+		return err
+	}
+	for i, element := range list {
+		err := tx.LSet(bucket, key, i, element)
```

Comment:
I don’t understand why you need to set all the items

Reply:
because we need all items have the same timestamp， include the item that list had before

Reply:
if you think it is correct or incorrect，tell me and I will pull a new version